### PR TITLE
Totally overhaul octane-node

### PR DESCRIPTION
**Pull request by salemhilal**
**This one's a big one, but a lot of it is generated code**. Read this PR commit-by-commit.

## TL;DR:

- I updated the generator to the newest version.
- I turned on a flag that made it use an object to pass in API parameters, rather than positional function arguments.
- I cleaned up the `src/resources/*.ts` files to match these new generated functions.
- I updated the tests to make sure they still pass.
- I fixed any remaining problems that we were using `sed` to fix.
- I installed the generator as an `npm` package, which uses a universal Java binary under the hood. In other words, no more need for Docker.
- I documented all of this in a brand new `CHANGELOG.md` and in `README.md`.

## What I need from *you*

- Can you look at the CHANGELOG and the README and make sure they make sense?
- Can you glance at the `src/resources/*.ts` files and make sure they seem correct?

### CODE REVIEW

**Review by akshalaby — APPROVED** [review]
Thank you Salem for the detailed commits!  Happy with the overall changes.  Didn't notice any issues in particular, is there anything I should dig deeper into?

**Comment by salemhilal** [comment]
@akshalaby the biggest things are honestly the CHANGELOG and the README to see if they make sense. If nothing jumped out to you, I'm gonna merge this and bump the release version tomorrow morning, which should be pretty safe to do.
